builder/node.go

```diff
@@ -34,9 +34,27 @@ func (b *Builder) Nodes() []Node {
 	return b.nodes
 }
 
+// AvailableNodes returns only nodes that are available.
+func (b *Builder) AvailableNodes() ([]Node, error) {
+	out := make([]Node, 0, len(b.nodes))
+	err := errors.Errorf("no drivers found")
+	for _, n := range b.nodes {
+		if n.Err == nil && n.Driver != nil {
+			out = append(out, n)
+		}
+		if n.Err != nil {
+			err = n.Err
+		}
+	}
+	if len(out) > 0 {
+		return out, nil
+	}
+	return nil, err
+}
+
 // LoadNodes loads and returns nodes for this builder.
 // TODO: this should be a method on a Node object and lazy load data for each driver.
-func (b *Builder) LoadNodes(ctx context.Context, withData bool) (_ []Node, err error) {
+func (b *Builder) LoadNodes(ctx context.Context, withData, onlyAvailable bool) (_ []Node, err error) {
```

Comment:
Ideally, I'd like to not modify this to take another boolean.

`AvailableNodes` can be renamed to `LoadAvailableNodes` - then `LoadNodes` and `LoadAvailableNodes` can call `LoadNodes` internally, and do the filtering on that directly.

Reply:
I think that makese sense, thanks 👍

Reply:
PTAL [32c9bf0](https://github.com/docker/buildx/pull/1466/commits/32c9bf0683cf25f2d472c2fe939dce7daf9d774e)